api/src/main/java/eu/darkbot/api/extensions/MapGraphics.java

```diff
@@ -2,114 +2,164 @@
     }
 
-    default void setFont(Font font) {
-        getGraphics2D().setFont(font);
+    default void drawOval(int x, int y, int size, boolean fill) {
+        drawOval(x, y, size, size, fill);
     }
 
-    default void drawRect(Point pos, boolean fill, int width, int height) {
-        if (fill) getGraphics2D().fillRect(pos.x(), pos.y(), width, height);
-        else getGraphics2D().drawRect(pos.x(), pos.y(), width, height);
+    default void drawOval(Point point, int width, int height, boolean fill) {
+        drawOval(point.x(), point.y(), width, height, fill);
     }
 
-    default void drawRect(Locatable pos, boolean fill, int width, int height) {
-        Point p = translate(pos);
-        p = Point.of(p.x() - Math.round(width / 2d), p.y() - Math.round(height / 2d));
-
-        drawRect(p, fill, width, height);
+    default void drawOval(Locatable loc, int width, int height, boolean fill) {
+        drawOval((int) (toScreenPointX(loc.getX()) - Math.round(width / 2d)),
+                (int) (toScreenPointY(loc.getY()) - Math.round(height / 2d)), width, height, fill);
     }
 
-    default void drawRect(Locatable pos, boolean fill, int size) {
-        drawRect(pos, fill, size, size);
+    default void drawOval(Locatable loc, int size, boolean fill) {
+        drawOval(loc, size, size, fill);
     }
 
-    default void drawOval(Point pos, boolean fill, int width, int height) {
-        if (fill) getGraphics2D().fillOval(pos.x(), pos.y(), width, height);
-        else getGraphics2D().drawOval(pos.x(), pos.y(), width, height);
+    /**
+     * Draws polygon
+     */
+    default void drawPoly(PolyType type, @NotNull Point... points) {
+        int[] xPoints = new int[points.length];
+        int[] yPoints = new int[points.length];
+        for (int i = 0; i < points.length; i++) {
+            xPoints[i] = points[i].x();
+            yPoints[i] = points[i].y();
+        }
+
+        if (type == PolyType.DRAW_POLYGON) getGraphics2D().drawPolygon(xPoints, yPoints, points.length);
+        else if (type == PolyType.FILL_POLYGON) getGraphics2D().fillPolygon(xPoints, yPoints, points.length);
+        else if (type == PolyType.DRAW_POLYLINE) getGraphics2D().drawPolyline(xPoints, yPoints, points.length);
     }
 
-    default void drawOval(Locatable pos, boolean fill, int size) {
-        drawOval(pos, fill, size, size);
+    default void drawPoly(PolyType type, @NotNull Locatable... positions) {
+        Point[] points = new Point[positions.length];
+        for (int i = 0; i < positions.length; i++)
+            points[i] = toScreenPoint(positions[i]);
+
+        drawPoly(type, points);
     }
 
-    default void drawOval(Locatable pos, boolean fill, int width, int height) {
-        Point p = translate(pos);
-        p = Point.of(p.x() - Math.round(width / 2d), p.y() - Math.round(height / 2d));
+    @SuppressWarnings("SuspiciousToArrayCall")
+    default void drawPoly(PolyType type, @NotNull Collection<?> positions) {
```

Comment:
This is a no-no.
Specify the type of the collection clearly, and if you want to support both point and locatable make 2 methods and change the signature, but you can't just take in a collection of whatever, it's not clear